pkg/controller/catalog/storage/storage_util.go

```diff
@@ -45,66 +45,47 @@ func (e serviceNotFound) Error() string {
 //
 // If Plan is not given and there's only one plan for a given service, we'll
 // choose that.
-func GetServicePlanInfo(storage Storage, service string, plan string) (string, string, string, error) {
-	s, err := storage.GetServiceClass(service)
+func GetServicePlanInfo(storage ServiceClassStorage, service string, plan string) (string, string, string, error) {
+	s, err := storage.Get(service)
 	if err != nil {
 		return "", "", "", err
 	}
 	// No plan specified and only one plan, use it.
 	if plan == "" && len(s.Plans) == 1 {
-		log.Printf("Found Service Plan GUID as %s for %s : %s", s.Plans[0].ID, service, s.Plans[0].Name)
-		return s.ID, s.Plans[0].ID, s.Plans[0].Name, nil
+		planID := s.Plans[0].CFGUID
+		planName := s.Plans[0].Name
+		log.Printf("Found Service Plan GUID as %s for %s : %s", planID, service, planName)
+		return s.CFGUID, planID, planName, nil
 	}
 	for _, p := range s.Plans {
 		if p.Name == plan {
-			fmt.Printf("Found Service Plan GUID as %s for %s : %s", p.ID, service, plan)
-			return s.ID, p.ID, p.Name, nil
+			planID := p.CFGUID
+			log.Printf("Found Service Plan GUID as %s for %s : %s", planID, service, plan)
+			return s.CFGUID, planID, p.Name, nil
 		}
 	}
 	return "", "", "", servicePlanNotFound{service, plan}
 }
 
 // GetBrokerByServiceClass returns the broker which serves a particular service
 // class.
-func GetBrokerByServiceClass(storage Storage, id string) (*model.ServiceBroker, error) {
+func GetBrokerByServiceClass(
+	brokerStorage BrokerStorage,
+	svcClassStorage ServiceClassStorage,
```

Comment:
Let's try to pass in the minimal API surface here. Passing `storage.Storage` requires a lot more API that this func needs.

Passing in a heterogeneous mix is a real concern, and we can solve by creating a composed interface and passing that:

```go
type BrokerServiceClassStorage interface {
  Brokers() BrokerStorage
  ServiceClasses() ServiceClassStorage
}
```